src/hints/mod.rs

```diff
@@ -336,3 +339,26 @@ pub fn breakpoint(
     println!("\n-----------END BREAKPOINT-----------\n");
     Ok(())
 }
+
+#[allow(unused)]
+const IS_ON_CURVE: &str = "ids.is_on_curve = (y * y) % SECP_P == y_square_int";
+pub fn is_on_curve(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let y: BigInt = exec_scopes.get("y")?;
+    // TODO: should y_square_int be calculated or should caller put it in scope for us?
+    let y_square_int: BigInt = exec_scopes.get("y_square_int")?;
+    // TODO: should SECP_P be expected to be in scope, or should it exist implicitly (e.g. as a
+    // constant)?
+    let secp_p: BigInt = exec_scopes.get("SECP_P")?;
+
+    let is_on_curve = (y.clone() * y) % secp_p == y_square_int;
```

Comment:
Is that `clone()` required? Is `BigInt` not a `Copy` type?

Reply:
`BigInt` doesn't `impl Copy` probably because it is arbitrarily large, so `clone()` is required.